ibm/service/cos/data_source_ibm_cos_bucket.go

```diff
@@ -851,24 +851,21 @@ func dataSourceIBMCosBucketRead(d *schema.ResourceData, meta interface{}) error
 	getBucketWebsiteConfigurationInput := &s3.GetBucketWebsiteInput{
 		Bucket: aws.String(bucketName),
 	}
-
-	outputBucketWebsite, err := s3Client.GetBucketWebsite(getBucketWebsiteConfigurationInput)
-	var outputBucketWebsiteptr *s3.WebsiteConfiguration
-	outputBucketWebsiteptr = (*s3.WebsiteConfiguration)(outputBucketWebsite)
-	if err != nil && !strings.Contains(err.Error(), "AccessDenied: Access Denied") {
+	outputwebsite, err := s3Client.GetBucketWebsite(getBucketWebsiteConfigurationInput)
+	var outputptr *s3.WebsiteConfiguration
+	outputptr = (*s3.WebsiteConfiguration)(outputwebsite)
+	if err != nil && !strings.Contains(err.Error(), "AccessDenied: Access Denied") && !strings.Contains(err.Error(), "The specified bucket does not have a website configuration") {
 		return err
 	}
-
-	if outputBucketWebsite != nil {
-		websiteConfiguration := flex.WebsiteConfigurationGet(outputBucketWebsiteptr)
+	if outputwebsite.IndexDocument != nil || outputwebsite.RedirectAllRequestsTo != nil {
```

Comment:
outputBucketWebsite != nil also do we needto check?

Reply:
instead of  outputBucketWebsite != nil I am checking  outputwebsite.IndexDocument != nil . because outputBucketwebsite is returned empty when no website configuration {} so the outputBucketWebsite != nil check was passing which we don't want.